packages/java/parser-jvm-plugin-backbone/src/main/java/dev/hilla/parser/plugins/backbone/MethodPlugin.java

```diff
@@ -22,44 +26,59 @@ public final class MethodPlugin extends AbstractPlugin<PluginConfiguration> {
-                    .getSource();
-            var methods = endpointCls.getMethodsStream()
+        var node = nodeDependencies.getNode();
+        if (node instanceof EndpointNode
+                || node instanceof EndpointExposedNode) {
+            var endpointCls = (ClassInfoModel) node.getSource();
+            var methodNodes = endpointCls.getMethodsStream()
                     .filter(MethodInfoModel::isPublic)
                     .<Node<?, ?>> map(MethodNode::of);
-            return nodeDependencies.appendChildNodes(methods);
+            return nodeDependencies.appendChildNodes(methodNodes);
         }
+
         return nodeDependencies;
     }
 
     @Override
     public void enter(NodePath<?> nodePath) {
-        var node = nodePath.getNode();
-        var parentNode = nodePath.getParentPath().getNode();
-        if (node instanceof MethodNode && parentNode instanceof EndpointNode) {
-            var methodNode = (MethodNode) node;
-            var endpointNode = (EndpointNode) parentNode;
-            methodNode.getTarget()
-                    .post(createOperation(endpointNode, methodNode));
+        if (!(nodePath.getNode() instanceof MethodNode)) {
+            return;
+        }
+
+        var methodNode = (MethodNode) nodePath.getNode();
+
+        var endpointParent = findClosestEndpoint(nodePath);
+        if (endpointParent.isEmpty()) {
+            return;
         }
+
+        var endpointNode = (EndpointNode) endpointParent.get();
+
+        methodNode.getTarget().post(createOperation(endpointNode, methodNode));
     }
 
     @Override
     public void exit(NodePath<?> nodePath) {
-        var node = nodePath.getNode();
-        var parentNode = nodePath.getParentPath().getNode();
-        var grandParentNode = nodePath.getParentPath().getParentPath()
-                .getNode();
-        if (node instanceof MethodNode && parentNode instanceof EndpointNode
-                && grandParentNode instanceof RootNode) {
-            var endpointName = ((EndpointNode) parentNode).getTarget()
-                    .getName();
-            var methodName = ((MethodNode) node).getSource().getName();
-            ((RootNode) grandParentNode).getTarget().path(
-                    String.format("/%s/%s", endpointName, methodName),
-                    ((MethodNode) node).getTarget());
+        if (!(nodePath.getNode() instanceof MethodNode)) {
+            return;
+        }
+
+        var methodNode = (MethodNode) nodePath.getNode();
+
+        var endpointParent = findClosestEndpoint(nodePath);
+        if (endpointParent.isEmpty()) {
+            return;
         }
+
+        var endpointNode = (EndpointNode) endpointParent.get();
+
+        var endpointName = endpointNode.getTarget().getName();
+        var methodName = methodNode.getSource().getName();
+
+        var rootNode = (RootNode) nodePath.getRootPath().getNode();
+        rootNode.getTarget().path(
+                String.format("/%s/%s", endpointName, methodName),
```

Comment:
```suggestion
                '/' + endpointName + '/' + methodName,
```
will allow you to gain an extra nanosecond 😉